### PR TITLE
feat: check whether port is used before server creation

### LINKED ISSUE #4178: Conflicts with applications running first on the same port
### Version

[v2.0.0](https://github.com/nuxt.js/releases/tag/v2.0.0)

### Reproduction link

[https://github.com/ohjihwan-nbt/default-nuxt-structure](https://github.com/ohjihwan-nbt/default-nuxt-structure)

### Steps to reproduce

1. Create node application that makes HTTP request of port 4000 and execute
- Execute with node command after creating the following file

```javascript
const http = require ('http')

const server = http.createServer ()
server.on ('request', (req, res) => {
   res.end ('OK')
}}

server.listen (4000)
```

2. Create and run the Nuxt application that makes http request on port 4000
- Run `npm run dev` after the repository clone: https://github.com/ohjihwan-nbt/default-nuxt-structure

3. Check the results after requesting http: // localhost: 4000 from the browser
- Check 'OK' string response created in node application executed in # 1

### What is expected ?

I expect that the nuxt application will not be processed after the following error message after running the application.

Error: listen EADDRINUSE ::: 4000

### What is actually happening?

After running `npm run dev`, I see the following message.

READY Listening on http: // localhost: 4000

### Additional comments?

I think there should be a check to see if the port is in use elsewhere before appServer is listening and also think it is correct to show a message such as ERROR rather than READY. 

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8022">#c8022</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by fureweb-com**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

this commit resolves #4178

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Comment by pimlie** [comment]
Why can't you just listen for an error event on the existing server instance? Like the Node docs describe here: https://nodejs.org/api/net.html#net_server_listen

**Comment by fureweb-com** [comment]
> Why can't you just listen for an error event on the existing server instance? Like the Node docs describe here: https://nodejs.org/api/net.html#net_server_listen

Sorry about that. Honestly I don't know well about net api, so I just use that way.

File: packages/core/src/nuxt.js [hunk 1]
```diff
@@ -136,6 +136,15 @@ export default class Nuxt {
     this.readyMessage = null
   }
 
+  isPortInUse(port) {
+    return new Promise((resolve, reject) => {
+      const tester = require('net').createServer()
+        .once('error', err => (err.code === 'EADDRINUSE' ? resolve(true) : reject(err)))
+        .once('listening', () => tester.once('close', () => resolve(false)).close())
```
Comment (galvez) [inline packages/core/src/nuxt.js]: @fureweb-com seems we've got a problem with this line. It's not enough time for the port to be cleared and reused, as the failing tests suggest.

**Review by galvez — CHANGES_REQUESTED** [review]

File: packages/core/src/nuxt.js (hunk 1 above)
Comment (galvez) [reply packages/core/src/nuxt.js]: Also probably a good idea to follow @pimlie's idea.

**Comment by fureweb-com** [comment]
When I ran it on MacOS, it was Okay.
I think this problem is not related to nuxt.js so I close this issue. sorry for the confusion.

**Pull request closed without merging**